ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/linearoptimisation/algorithms/linearproblem/OpenRaoMPSolver.java

```diff
@@ -140,30 +170,65 @@ public LinearProblemStatus solve() {
         return convertResultStatus(mpSolver.solve(solveConfiguration));
     }
 
-    private static LinearProblemStatus convertResultStatus(MPSolver.ResultStatus status) {
-        switch (status) {
-            case OPTIMAL:
-                return LinearProblemStatus.OPTIMAL;
-            case ABNORMAL:
-                return LinearProblemStatus.ABNORMAL;
-            case FEASIBLE:
-                return LinearProblemStatus.FEASIBLE;
-            case UNBOUNDED:
-                return LinearProblemStatus.UNBOUNDED;
-            case INFEASIBLE:
-                return LinearProblemStatus.INFEASIBLE;
-            case NOT_SOLVED:
-                return LinearProblemStatus.NOT_SOLVED;
-            default:
-                throw new NotImplementedException(String.format("Status %s not handled.", status));
-        }
+    static LinearProblemStatus convertResultStatus(MPSolver.ResultStatus status) {
+        return switch (status) {
+            case OPTIMAL -> LinearProblemStatus.OPTIMAL;
+            case ABNORMAL -> LinearProblemStatus.ABNORMAL;
+            case FEASIBLE -> LinearProblemStatus.FEASIBLE;
+            case UNBOUNDED -> LinearProblemStatus.UNBOUNDED;
+            case INFEASIBLE -> LinearProblemStatus.INFEASIBLE;
+            case NOT_SOLVED -> LinearProblemStatus.NOT_SOLVED;
+            default -> throw new OpenRaoException(String.format("Status %s not handled.", status));
+        };
     }
 
     public int numVariables() {
-        return mpSolver.numVariables();
+        return variables.size();
     }
 
     public int numConstraints() {
-        return mpSolver.numConstraints();
+        return constraints.size();
+    }
+
+    public boolean minimization() {
```

Comment:
I believe the naming convention for boolean getters is: isMinimization() / isMaximization()